### PR TITLE
Implement GitHub template files

### LINKED ISSUE #2460: structured documentation
The useful information in [doc](https://github.com/NebulousLabs/Sia/tree/master/doc) is difficult to navigate, since it's just a bunch of files. Please tie this into modern documentation tools such as [readthedocs][rtd]. This should be trivial to accomplish with a simple `index.rst` and automatic conversion of the `md` files into `html` on the doco platform you choose, and will produce a beautifully formatted version with easy navigation and clear structure.

If GitHub is the focal point for Sia development, it would also help to make use of the platform's tools for making contributing guidelines, easily discoverable, as well as providing templates for new issues and
pull requests to help guide the iterative process of accepting community input. Details are in the Insights
tab under Community.

[rtd]: https://readthedocs.io

### PR DESCRIPTION
**Pull request by tbenz9**
This commit implements very basic contributing and issue template files.  Template files are displayed in a banner when a user tries to perform an action such as create a new issue.  More details can be found here: https://github.com/blog/2111-issue-and-pull-request-templates

Fixes #2460

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+**Any useful logs**
+
+
+
+Feature Request
+===============
+
+Clearly describe your feature request and why it would benefit {{.RepoName}}
```
Comment (lukechampine) [inline .github/ISSUE_TEMPLATE.md]: Sia?

Maybe also link to the roadmap or similar to prevent duplicate requests

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Interested in contributing to SIA?
```
Comment (lukechampine) [inline .github/CONTRIBUTING.md]: Sia is never all-caps

**Comment by tbenz9** [comment]
There is also an option to include a PR template.  I didn't think it was necessary, but if you are interested I would be happy to reconsider it.

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+* OS:
+
+
+**Any useful logs**
+
+
+
+Feature Request
+===============
+Check to see if the feature is on the roadmap.  If it is not on the roadmap then clearly describe your feature request and why it would benefit {{.RepoName}}
```
Comment (lukechampine) [inline .github/ISSUE_TEMPLATE.md]: is `{{.RepoName}}` automatically filled in? Is there a reason to not just use Sia?

**Pull request merged**